Drop reapers

soltysh opened the pull request:
/assign @deads2k @juanvallejo 

**Release note**:
```release-note
kubectl delete does not use reapers for removing objects anymore, but relies on server-side GC entirely
```

soltysh committed:
Remove kubectl reapers

soltysh commented:
Only last commit matters, b/c it waits for https://github.com/kubernetes/kubernetes/pull/59851 to merge first.

deads2k commented on pkg/kubectl/cmd/apply_test.go:
```diff
@@ -1200,176 +1194,177 @@ func checkPatchString(t *testing.T, req *http.Request) {
-		})
-	}
-}
+// TODO: fix this test
```
um...  Yeah. you can't merge it like this.

soltysh replied:
Duh :tong

deads2k commented:
/hold

don't want to accidentally merge with tests commented out.

soltysh commented:
/hold cancel
Rebased and test fixed. @deads2k @juanvallejo ptal

deads2k commented on pkg/kubectl/cmd/apply.go:
```diff
@@ -790,17 +769,15 @@ func (p *patcher) patch(current runtime.Object, modified []byte, source, namespa
 		return modified, nil, err
 	}
-	err = wait.PollImmediate(kubectl.Interval, p.timeout, func() (bool, error) {
+	if err := wait.PollImmediate(1*time.Second, p.timeout, func() (bool, error) {
```
// TODO use wait

deads2k commented on pkg/kubectl/cmd/delete.go:
```diff
@@ -301,14 +236,18 @@ func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
+		}
 		options.PropagationPolicy = &policy
-		return o.deleteResource(info, options)
+		// TODO: nilebox's https://github.com/kubernetes/kubernetes/pull/63695
```
out of date comment

deads2k commented on pkg/kubectl/cmd/drain.go:
```diff
@@ -594,7 +593,7 @@ func (o *DrainOptions) evictPods(pods []corev1.Pod, policyGroupVersion string, g
 			}
 			podArray := []corev1.Pod{pod}
-			_, err = o.waitForDelete(podArray, kubectl.Interval, time.Duration(math.MaxInt64), true, getPodFn)
+			_, err = o.waitForDelete(podArray, 2*time.Second, time.Duration(math.MaxInt64), true, getPodFn)
```
didn't I see a one second above?

deads2k commented on pkg/kubectl/scale_test.go:
```diff
@@ -626,3 +630,54 @@ func TestGenericScale(t *testing.T) {
 	}
 }
+
+func createFakeScaleClient(resource string, resourceName string, replicas int, errorsOnVerb map[string]*kerrors.StatusError) *fakescale.FakeScaleClient {
```
Where is this from and used?

deads2k replied:
> Where is this from and used?

an from delete, used in scale.

deads2k commented:
nits

/lgtm

soltysh commented:
cmd-test fixed, re-applying the label.

soltysh commented:
/hold 
I'm going to add some debug to find the root cause, and I don't want this be merged by accident.

soltysh committed:
Handle DaemonSet removal the old way

soltysh commented:
/hold cancel

deads2k commented on hack/make-rules/test-cmd-util.sh:
```diff
@@ -1478,7 +1478,7 @@ __EOF__
   kubectl delete rs frontend "${kube_flags[@]}"
   kubectl delete rc frontend "${kube_flags[@]}"
-  kubectl delete ds bind "${kube_flags[@]}"
+  kubectl delete -f hack/testdata/rollingupdate-daemonset.yaml "${kube_flags[@]}"
```
revert

soltysh replied:
Reverted.

deads2k commented:
nit

/lgtm

soltysh commented:
Nit fixed, re-tagging.

soltysh committed:
Remove initializer test
Initializers are alpha, broken and a subject for removal. They don't
work well with finalizers and the previous hack present in deployment
and replicaset reapers was just hiding this problem.

soltysh commented:
Hack in deployment reaper: https://github.com/kubernetes/kubernetes/blob/8306b0b1a4915d4b4e44f475aedf4345d592a7b7/pkg/kubectl/delete.go#L425-L429
